packages/dashboard/src/pages/Personas/hooks/useConnectSocialNetwork.ts

```diff
@@ -0,0 +1,14 @@
+import { useAsyncFn } from 'react-use'
+import { Services } from '../../../API'
+
+export interface SocialNetworkProvider {
+    internalName: string
+    network: string
+    connected: boolean
+}
+
+export function useConnectSocialNetwork() {
+    return useAsyncFn(async (identifier: string, provider: SocialNetworkProvider) => {
+        return Services.SocialNetwork.connectSocialNetwork(identifier, provider)
+    })
```

Comment:
It seems `connectSocialNetwork` returns `Promise<void>`. Why not just call it directly?

Reply:
The `loading` state may be used later to display the loading indicator